### PR TITLE
Documentation how to make 'step series' closes #698

### LINKED ISSUE #698: Add step series chart
In the d3fc vision there is a mention of step series. It looks like that:

http://www.highcharts.com/stock/demo/step-line

It is getting more popular and I have seen it in some of our designs. I think it should be very similar to line series, just rendered differently, so maybe a different renderer is what is needed.

### PR DESCRIPTION
**Pull request by bjedrzejewski**
Thanks to @jleft it turns out that we can already do a step series pretty easy, the documentation just wasnt there. I updated the documentation then.

### CODE REVIEW

**Comment by ColinEberhardt** [comment]
Thanks for this - I think it is very helpful for our users to inform them of some of the D3 features that we support (via rebind etc ...) but they might not be aware of.

However, we shouldn't reproduce large chunks of D3 documentation, what happens if this changes in future? How do we maintain our copy?

I'd instead link to the d3 interpolate documentation page where people can read about the interpolation modes.

Other than that, looks good to me :-)

File: site/src/components/series/line.md [hunk 1]
```diff
@@ -27,3 +48,53 @@ Which gives the following:
 
 This series has the same properties for specifying the x and y value accessors, and decorate as the [point series](#point).
 
+`Line` also exposes the `interpolate` property from the d3 library, which accepts 11 different interpolations:
```
Comment (jleft) [inline site/src/components/series/line.md]: I think there are actually 13 types of interpolation - `linear-closed` and `step` are missing from this list.

https://github.com/mbostock/d3/blob/v3.5.6/src/svg/line.js#L83-L98

**Comment by bjedrzejewski** [comment]
Will link the documentation and just mention couple of them for the examples.

**Comment by bjedrzejewski** [comment]
@jleft  @ColinEberhardt updated and rebased

**Comment by jleft** [comment]
:+1: Looks good :)

**Comment by jdunkerley** [comment]
Looks good to me. Another one for the multi example playground...

**Comment by ColinEberhardt** [comment]
Thanks @bjedrzejewski - just spotted something, the `basis` interpolation mode does not support tension. Rather than provide an exhaustive set of examples I've simplified just to highlight the step interpolation mode.

**Pull request merged**